lightly_studio/src/lightly_studio/core/add_videos.py

```diff
@@ -184,27 +208,46 @@ def _create_video_frame_samples(
             frame_number=decoded_index,
             frame_timestamp_s=frame_timestamp_s,
             frame_timestamp_pts=frame.pts if frame.pts is not None else -1,
-            parent_sample_id=video_sample_id,
+            parent_sample_id=context.video_sample_id,
         )
         samples_to_create.append(sample)
 
         # Process batch when it reaches SAMPLE_BATCH_SIZE
         if len(samples_to_create) >= SAMPLE_BATCH_SIZE:
             created_samples_batch = video_frame_resolver.create_many(
-                session=session,
+                session=context.session,
                 samples=samples_to_create,
-                dataset_id=dataset_id,
+                dataset_id=context.dataset_id,
             )
             created_sample_ids.extend(created_samples_batch)
             samples_to_create = []
 
     # Handle remaining samples for this video
     if samples_to_create:
         created_samples_batch = video_frame_resolver.create_many(
-            session=session,
+            session=context.session,
             samples=samples_to_create,
-            dataset_id=dataset_id,
+            dataset_id=context.dataset_id,
         )
         created_sample_ids.extend(created_samples_batch)
 
     return created_sample_ids
+
+
+def _configure_stream_threading(video_stream: VideoStream, num_decode_threads: int | None) -> None:
+    """Configure codec-level threading for faster decode when available."""
+    codec_context = getattr(video_stream, "codec_context", None)
+    if codec_context is None:
+        return
+
+    if num_decode_threads is None:
+        cpu_count = os.cpu_count() or 1
+        # Use half available cores but at least 1. Cap to prevent runaway usage.
+        num_decode_threads = max(1, min(cpu_count // 2 or 1, 16))
+
+    try:
+        codec_context.thread_type = ThreadType.AUTO
+        codec_context.thread_count = num_decode_threads
+    except av.AVError:
+        # Some codecs do not support threading—ignore silently.
+        pass
```

Comment:
This function can be nicely tested (by patching `os.cpu_count()` and passing a MagicMock as video_stream), let's add a test for it.